nvflare/apis/utils/reliable_message.py

```diff
@@ -580,6 +586,10 @@ def _send_request(
                 # the ack is a status report - check status
                 status = ack.get_header(HEADER_STATUS)
                 if status and status != STATUS_NOT_RECEIVED:
+                    if status == STATUS_DUP_REQUEST:
+                        cls.debug(fl_ctx, "Duplicate req status received")
+                        return ack
```

Comment:
No, should be "break" not "return ack" so that it will go to the cls._query_result().

Reply:
In that case, this block of code is not needed. Removed.